### PR TITLE
Make object contributions only show up in python files

### PR DESCRIPTION
**Pull request by apaku**
Things like the 2to3 or the analysis actions only make sense when working
with an actual python file. However by letting them be visible based on
the project nature these actions also end up in the context menus of any
other types of editors used for files within a python project. That means
an xml file happening to sit in a python project will show these actions as
well as a simple text file.

So instead let the actions only show up for resources that are of the
python content type.

### CODE REVIEW

File: plugins/com.python.pydev.analysis/plugin.xml [hunk 1]
```diff
@@ -177,7 +177,7 @@
                menubarPath="org.python.pydev.ui.actions.menu/pydev"
                tooltip="Force a code analysis for the selected elements (recursive)">
          </action>
-         <visibility><objectState name="projectNature" value="org.python.pydev.pythonNature"/></visibility>
+         <visibility><objectState name="contentTypeId" value="org.python.pydev.pythonfile"/></visibility>
```
Comment (fabioz) [inline plugins/com.python.pydev.analysis/plugin.xml]: Actually, those actions can be executed when a project/folder is selected... with this changes I don't think they'd work anymore right? The visibility should still enable them for any project/folder selected in a project with a pythonNature (for both cases).

Comment (apaku) [inline plugins/com.python.pydev.analysis/plugin.xml]: I had that same thought shortly after putting up the PR. I'll see wether that can be expressed, don't have too much experience with this kind of object contribution, and otherwise will discard the request.

Comment (apaku) [inline plugins/com.python.pydev.analysis/plugin.xml]: I experimented with this a bit, the existing objectState check has no sufficient test for something like "is a python file or a folder containing some python file". I've also experimented with changing the object class to PythonSourceFolder to enable it for python source folders (in addition to python files), but that does not seem to work very well for source folders added to a pydev project manually. At least the contribution does not show up there - but it does show up with a right-click on the project itself.

The last option I see (but haven't tried) is converting these actions to use the org.eclipse.ui.menus extension point for contribution. With that one could use command expressions to define the enablement/visibility and that supports a way to have code executed for such tests through propertyTester's.

Thats of course a bigger change so maybe not worth it for a corner case like ours...

**Comment by fabioz** [comment]
Well, I think the check should be: is this a python file (something with a python shebang even if not a .py file) or any folder inside a project with a Python nature... I think the only option is doing it through a way that supports propertyTesters, as for being worth it or not, I'll leave that up to you :)

**Comment by apaku** [comment]
I closed the PR now, failed to figure out an easy way to re-use the
existing action code and then ended up not getting my command to show up at
all. Given that this was more a 'make the UI a little cleaner' effort I'm
going to stop that here and not waste more time on it.
